src/stdlib_linalg_schur.fypp

```diff
@@ -0,0 +1,305 @@
+        complex(${rk}$), optional, intent(out), contiguous, target :: eigvals(:)
+        !> [optional] Provide pre-allocated workspace, size to be checked with schur_space
+        ${rt}$, optional, intent(inout), target :: storage(:)
+        !> [optional] Can A data be overwritten and destroyed?
+        logical(lk), optional, intent(in) :: overwrite_a        
+        !> [optional] State return flag. On error if not requested, the code will stop
+        type(linalg_state_type), optional, intent(out) :: err
+
+        ! Local variables
+        integer(ilp) :: m,n,mt,nt,ldvs,nvs,lde,lwork,sdim,info
+        logical(lk) :: overwrite_a_
+        logical(lk), target :: bwork_dummy(1),local_eigs
+        logical(lk), pointer :: bwork(:)
+        real(${rk}$), allocatable :: rwork(:)
+        ${rt}$, target :: vs_dummy(1,1)
+        ${rt}$, pointer :: vs(:,:),work(:),eigs(:)#{if rt.startswith('r')}#,eigi(:)#{endif}#
+        character :: jobvs,sort
+        type(linalg_state_type) :: err0
+        
+        ! Problem size
+        m  = size(a, 1, kind=ilp)
+        n  = size(a, 2, kind=ilp)        
+        mt = size(t, 1, kind=ilp)
+        nt = size(t, 2, kind=ilp)
+        
+        ! Validate dimensions
+        if (m/=n .or. m<=0 .or. n<=0) then
+            err0 = linalg_state_type(this, LINALG_VALUE_ERROR, 'Matrix A must be square: size(a)=',[m,n])
+            call linalg_error_handling(err0, err)
+            return
+        end if    
+        if (mt/=nt .or. mt/=n .or. nt/=n) then
+            err0 = linalg_state_type(this, LINALG_VALUE_ERROR, 'Matrix T must be square: size(T)=',[mt,nt], &
+                                                          'should be',[m,n])
+            call linalg_error_handling(err0, err)
+            return
+        end if                
+        
+        !> Copy data into the output array
+        t = a 
+        
+        ! Can A be overwritten? By default, do not overwrite
+        if (present(overwrite_a)) then 
+            overwrite_a_ = overwrite_a .and. n>=2    
+        else
+            overwrite_a_ = .false._lk
+        endif
+        
+        !> SORTING: no sorting options are currently supported
+        sort  = gees_sort_eigs(.false.)
+        sdim  = 0_ilp     
+        
+        if (sort/=GEES_NOT) then 
+            
+           allocate(bwork(n),source=.false.)
+        
+        else
+            
+           bwork => bwork_dummy 
+            
+        end if        
+        
+        !> Schur vectors
+        jobvs = gees_vectors(present(z))        
+        if (present(z)) then 
+            vs => z
+            
+            ldvs = size(vs, 1, kind=ilp)
+            nvs  = size(vs, 2, kind=ilp)
+            
+            if (ldvs<n .or. nvs/=n) then
+                err0 = linalg_state_type(this, LINALG_VALUE_ERROR, 'Schur vectors size=',[ldvs,nvs], &
+                                                              'should be n=',n)
+                goto 1
```

Comment:
The `goto 1` could be avoided, right (e.g., by adding the assigned `err` and a `return`)?
This would facilitate the understanding of the procedure.